src/libraries/System.Reflection/tests/MethodInfoTests.cs

```diff
@@ -763,6 +763,48 @@ public static void InvokeNullableEnumParameterNoDefault()
             Assert.Throws<ArgumentException>(() => method.Invoke(null, new object?[] { Type.Missing }));
         }
 
+        public static IEnumerable<object[]> MethodNameAndArgumetns()
+        {
+            yield return new object[] { nameof(Sample.DefaultString), "Hello", "Hi" };
+            yield return new object[] { nameof(Sample.DefaultNullString), null, "Hi" };
+            yield return new object[] { nameof(Sample.DefaultNullableInt), 3, 5 };
+            yield return new object[] { nameof(Sample.DefaultNullableEnum), YesNo.Yes, YesNo.No };
+        }
+
+        [Theory]
+        [MemberData(nameof(MethodNameAndArgumetns))]
+        public static void InvokeCopiesBackMissingArgument(string methodName, object defaultValue, object passingValue)
+        {
+            MethodInfo method = typeof(Sample).GetMethod(methodName);
+            object[] args = new object[] { Missing.Value };
+
+            Assert.Equal(defaultValue, method.Invoke(null, args));
+            Assert.Equal(defaultValue, args[0]);
+
+            args[0] = passingValue;
+
+            Assert.Equal(passingValue, method.Invoke(null, args));
+            Assert.Equal(passingValue, args[0]);
+
+            args[0] = null;
+            Assert.Null(method.Invoke(null, args));
+            Assert.Null(args[0]);
+        }
+
+        [Fact]
+        public static void InvokeCopiesBackMissingParameterAndArgument()
+        {
+            MethodInfo method = typeof(Sample).GetMethod(nameof(Sample.DefaultMissing));
+            object[] args = new object[] { Missing.Value };
+
+            Assert.Equal(Missing.Value, method.Invoke(null, args)); // Argument type matches with parameter type, therefore default value null will not copied back
```

Comment:
I do not think that this is the right behavior. Also, it is different from .NET 6 behavior.

Missing should be always replaced by the default value, even when the argument type itseft is Missing.

Reply:
This scenario goes through the type match path, therefore we will need to have a tweak in that path
https://github.com/dotnet/runtime/blob/312bd14178a8df5d1b0d54c0f5d7748eacf59ce4/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBase.cs#L176-L180

For me it is such an unimportant edge case scenario and current behavior is also makes sense to me, therefore I left it as is, but if you feel we need to change it I can do that

Reply:
The minimal tweak on that path should be very small - just need to add `&& argType != typeof(Missing)`.

The reflection is full of these edge cases. I think it is important that we have them fully speced and tested to avoid introducing accidental breaking changes as we are changing the implementation.